main/src/library/RichString.flix

```diff
@@ -69,59 +69,406 @@ mod RichString {
         case Span({text = String, fgColor = Color, bgColor = Color, styles = Set[Style]})
     }
 
+    ///
+    /// A record type representing a palette of colors.
+    ///
+    type alias ColorPalette = {
+        black   = Color,
+        gray    = Color,
+        red     = Color,
+        green   = Color,
+        yellow  = Color,
+        blue    = Color,
+        magenta = Color,
+        cyan    = Color,
+        white   = Color
+    }
+
     ///
     /// A trait for types that can be formatted as a `RichString`.
     ///
     pub trait Formattable[a] {
-        pub def format(x: a): RichString
+        ///
+        /// The associated effect of the Formattable which represents the effect of formatting.
+        ///
+        type Aef[a]: Eff = {}
+
+        pub def formatWithQuote(quote: Bool, x: a): RichString \ Formattable.Aef[a]
```

Comment:
why does the base function need the option for quotes?

Reply:
I think the color conventions should be listed in the doc comment

Reply:
so `withQuote` means that all strings in `x` are quoted?

Hmm it feels wrong to me. The above design also means that 
```
enum Example { case Foo }
..
Logger.info("Foo");
Logger.info(Example.Foo)
```
will just output
```
Foo
Foo
```